File: utils/start.js
const BashShell = require("../lib/BashShell.js");
const secrets = require("../secrets/secrets.js").getSecrets("./secrets/");

function myHandler(event) {
    console.log("SUBSHELL " + event.process + "  " + event.type);
    const printData = event.data.split("\n").map(ln => "    " + ln).join("\n");
    if (event.type === "err") {
        console.error(printData);
    } else {
        console.log(printData);
    }
}

if (secrets.USE_PROXY) {
    const ratholeProxy = new BashShell("RatholeProxy");
    ratholeProxy.handler = myHandler;
    ratholeProxy.send("./rathole client.toml");
}

const mainServer = new BashShell("MainServer");
mainServer.handler = myHandler;
mainServer.send("cd backend/main-server");
mainServer.send("node index.js");

const sandoxServer = new BashShell("SandboxServer");
sandoxServer.handler = myHandler;
sandoxServer.send("cd backend/sandbox-server");
sandoxServer.send("node index.js");

File: utils/start.dart
import 'dart:io';

import '../secrets/secrets.dart';

const HOTRELOAD = false;

void main() async {
    if (secrets.USE_PROXY) {
        final ratholeProcess = await Process.start(
            "./rathole", ["./secrets/client.toml"],
            mode: ProcessStartMode.inheritStdio
        );
    }

    Process.start(
        "bun", "run buildscript.js".split(" "),
        workingDirectory: "./frontend",
        mode: ProcessStartMode.inheritStdio
    );

    final mainServerProcess = await Process.start(
        "dart", "run ${HOTRELOAD ? "-r " : ""}backend/main-server/main.dart".split(" "),
        mode: ProcessStartMode.inheritStdio
    );

    final sandboxServerProcess = await Process.start(
        "dart", "run backend/sandbox-server/main.dart".split(" "),
        mode: ProcessStartMode.inheritStdio
    );

}
